salesforce.go

```diff
@@ -333,60 +323,46 @@ func (sf *Salesforce) DeleteOne(sObjectName string, record any) error {
 	return nil
 }
 
-func (sf *Salesforce) InsertCollection(sObjectName string, records any, batchSize int) error {
+func (sf *Salesforce) InsertCollection(sObjectName string, records any, batchSize int) ([]SalesforceResult, error) {
 	validationErr := validateCollections(*sf, records, batchSize)
 	if validationErr != nil {
-		return validationErr
+		return nil, validationErr
 	}
 
-	dmlErr := doInsertCollection(*sf.auth, sObjectName, records, batchSize)
-	if dmlErr != nil {
-		return dmlErr
-	}
-
-	return nil
+	salesforceError, dmlErr := doInsertCollection(*sf.auth, sObjectName, records, batchSize)
+	return salesforceError, dmlErr
 }
 
-func (sf *Salesforce) UpdateCollection(sObjectName string, records any, batchSize int) error {
+func (sf *Salesforce) UpdateCollection(sObjectName string, records any, batchSize int) ([]SalesforceResult, error) {
 	validationErr := validateCollections(*sf, records, batchSize)
 	if validationErr != nil {
-		return validationErr
+		return nil, validationErr
 	}
 
-	dmlErr := doUpdateCollection(*sf.auth, sObjectName, records, batchSize)
-	if dmlErr != nil {
-		return dmlErr
-	}
-
-	return nil
+	salesforceError, dmlErr := doUpdateCollection(*sf.auth, sObjectName, records, batchSize)
+	return salesforceError, dmlErr
 }
 
-func (sf *Salesforce) UpsertCollection(sObjectName string, externalIdFieldName string, records any, batchSize int) error {
+func (sf *Salesforce) UpsertCollection(sObjectName string, externalIdFieldName string, records any, batchSize int) ([]SalesforceResult, error) {
 	validationErr := validateCollections(*sf, records, batchSize)
 	if validationErr != nil {
-		return validationErr
+		return nil, validationErr
 	}
 
-	dmlErr := doUpsertCollection(*sf.auth, sObjectName, externalIdFieldName, records, batchSize)
-	if dmlErr != nil {
-		return dmlErr
-	}
+	salesforceError, dmlErr := doUpsertCollection(*sf.auth, sObjectName, externalIdFieldName, records, batchSize)
+	return salesforceError, dmlErr
 
-	return nil
 }
 
-func (sf *Salesforce) DeleteCollection(sObjectName string, records any, batchSize int) error {
+func (sf *Salesforce) DeleteCollection(sObjectName string, records any, batchSize int) ([]SalesforceResult, error) {
 	validationErr := validateCollections(*sf, records, batchSize)
 	if validationErr != nil {
-		return validationErr
+		return nil, validationErr
 	}
 
-	dmlErr := doDeleteCollection(*sf.auth, sObjectName, records, batchSize)
-	if dmlErr != nil {
-		return dmlErr
-	}
+	salesforceError, dmlErr := doDeleteCollection(*sf.auth, sObjectName, records, batchSize)
+	return salesforceError, dmlErr
```

Comment:
as you do nothing with salesforceError, dmlErr

I would use

```suggestion
	return doDeleteCollection(*sf.auth, sObjectName, records, batchSize)
```

Reply:
Sure, makes sense